app/controllers/CourseController.scala

```diff
@@ -0,0 +1,77 @@
+package controllers
+
+import models.Course
+import models.JsonFormats._
+import play.api.libs.json.Json
+import play.api.mvc.{AbstractController, ControllerComponents}
+import play.modules.reactivemongo.{MongoController, ReactiveMongoApi, ReactiveMongoComponents}
+import javax.inject.Inject
+
+import scala.concurrent.{ExecutionContext, Future}
+import reactivemongo.bson.BSONDocument
+import reactivemongo.play.json._
+import collection._
+import reactivemongo.api.Cursor
+import javax.inject.Singleton
+import reactivemongo.api.commands.WriteResult
+
+
+@Singleton
+class CourseController @Inject()(val components: ControllerComponents, val reactiveMongoApi: ReactiveMongoApi) extends AbstractController(components)
+  with MongoController with ReactiveMongoComponents {
+
+  implicit def ec: ExecutionContext = components.executionContext
+
+  def collection: Future[JSONCollection] = database.map(
+    _.collection[JSONCollection]("courses"))
+
+  def allcourses() = Action.async {
+    find().map { courses =>
+      Ok(Json.toJson(courses))
+    }
+  }
+
+  //helper to move to CourseRepository
+  def find() = {
+    val cursor: Future[Cursor[Course]] = collection.map {
```

Comment:
The controllers shouldn't contains methods related to mongo or other db. Create another package and respective classes containing methods `find()`, `create (course: Course)`, `findid(id: Long)`